clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java

```diff
@@ -1017,6 +1023,153 @@ public void testStickyBatches() throws Exception {
         assertEquals(appends, 2 * expectedAppends);
     }
 
+    @Test
+    public void testUniformBuiltInPartitioner() throws Exception {
```

Comment:
This and testAdaptiveBuiltInPartitioner are new additions, the rest is just tweaks to reflect changes in function signature.